core/capabilities/webapi/target/capability.go

```diff
@@ -0,0 +1,168 @@
+package target
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"strings"
+	"sync"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/capabilities"
+	"github.com/smartcontractkit/chainlink-common/pkg/logger"
+	"github.com/smartcontractkit/chainlink-common/pkg/types/core"
+	"github.com/smartcontractkit/chainlink-common/pkg/values"
+	"github.com/smartcontractkit/chainlink/v2/core/capabilities/validation"
+	"github.com/smartcontractkit/chainlink/v2/core/services/gateway/handlers/webcapabilities"
+)
+
+const ID = "web-api-target@1.0.0"
+
+var _ capabilities.TargetCapability = &Capability{}
+
+var capabilityInfo = capabilities.MustNewCapabilityInfo(
+	ID,
+	capabilities.CapabilityTypeTarget,
+	"A target that sends HTTP requests to external clients via the Chainlink Gateway.",
+)
+
+// Capability is a target capability that sends HTTP requests to external clients via the Chainlink Gateway.
+type Capability struct {
+	capabilityInfo        capabilities.CapabilityInfo
+	connectorHandler      *ConnectorHandler
+	lggr                  logger.Logger
+	registry              core.CapabilitiesRegistry
+	config                Config
+	registeredWorkflows   map[string]struct{}
+	registeredWorkflowsMu sync.Mutex
```

Comment:
_nit_ RWMutex

Reply:
frequencies of read and write are quite similar